internal/strace/strace.go

```diff
@@ -217,13 +219,13 @@ func (r *Result) parseEnterSyscall(syscall, args string, logger *zap.SugaredLogg
 			// Save the contents between the first and last quote.
 			writeBuffer = args[firstQuoteIndex+1 : lastQuoteIndex]
 		}
-		logger.Debugw("write", "path", path, "size", bytesWritten)
-		r.recordFileWrite(path, []byte(writeBuffer), bytesWritten)
+		slog.DebugContext(ctx, "write", "path", path, "size", bytesWritten)
+		r.recordFileWrite(ctx, path, []byte(writeBuffer), bytesWritten)
 	}
 	return nil
 }
 
-func (r *Result) parseExitSyscall(syscall, args string, logger *zap.SugaredLogger) error {
+func (r *Result) parseExitSyscall(ctx context.Context, syscall, args string) error {
 	switch syscall {
```

Comment:
Does this allow changing the logging destination just from the context? I added the logger argument due to #798, is that behaviour still possible with passing just a context object?

Reply:
Ah. Good point. I'll fix this.